modules/rostests/apitests/ntdll/LdrLoadDll.c

```diff
@@ -0,0 +1,164 @@
+    StartSeh()
+        BaseAddress = InvalidPointer;
+        Status = LdrLoadDll(NULL, NULL, &DllName, &BaseAddress);
+        ok(Status == STATUS_SUCCESS, "Status = %lx\n", Status);
+        ok(BaseAddress != NULL && BaseAddress != InvalidPointer, "BaseAddress = %p\n", BaseAddress);
+        if (NT_SUCCESS(Status))
+        {
+            BaseAddress2 = InvalidPointer;
+            Status = LdrLoadDll(NULL, NULL, &DllName, &BaseAddress2);
+            ok(Status == STATUS_SUCCESS, "Status = %lx\n", Status);
+            ok(BaseAddress2 == BaseAddress, "BaseAddress2 = %p, expected %p\n", BaseAddress2, BaseAddress);
+            LdrUnloadDll(BaseAddress);
+        }
+    EndSeh(STATUS_SUCCESS);
+
+    RtlInitUnicodeString(&DllName, L"advapi32.dll");
+    StartSeh()
+        Status = LdrLoadDll(L"\\SystemRoot\\System32", NULL, &DllName, NULL);
+    EndSeh(STATUS_ACCESS_VIOLATION);
+
+    StringCchPrintfW(szPath, _countof(szPath), L"%s\\advapi32", szSysDir);
+
+    RtlInitUnicodeString(&DllName, szPath);
+    StartSeh()
+        Status = LdrLoadDll(szWinDir, NULL, &DllName, &BaseAddress);
+        ok(Status == STATUS_SUCCESS, "Status = %lx\n", Status);
+        if (NT_SUCCESS(Status))
+            LdrUnloadDll(BaseAddress);
+    EndSeh(STATUS_SUCCESS);
+
+    StringCchPrintfW(szPath, _countof(szPath), L"%s\\advapi32.dll", szSysDir);
+
+    RtlInitUnicodeString(&DllName, szPath);
+    StartSeh()
+        Status = LdrLoadDll(szWinDir, NULL, &DllName, &BaseAddress);
+        ok(Status == STATUS_SUCCESS, "Status = %lx\n", Status);
+        if (NT_SUCCESS(Status))
+            LdrUnloadDll(BaseAddress);
+    EndSeh(STATUS_SUCCESS);
+
+    StringCchPrintfW(szPath, _countof(szPath), L"%s/advapi32", szSysDir);
+
+    RtlInitUnicodeString(&DllName, szPath);
+    StartSeh()
+        Status = LdrLoadDll(szWinDir, NULL, &DllName, &BaseAddress);
+        ok(Status == STATUS_SUCCESS, "Status = %lx\n", Status);
+        if (NT_SUCCESS(Status))
+            LdrUnloadDll(BaseAddress);
+    EndSeh(STATUS_SUCCESS);
+
+    StringCchPrintfW(szPath, _countof(szPath), L"%s/advapi32.dll", szSysDir);
+
+    RtlInitUnicodeString(&DllName, szPath);
+    StartSeh()
+        Status = LdrLoadDll(szWinDir, NULL, &DllName, &BaseAddress);
+        ok(Status == STATUS_SUCCESS, "Status = %lx\n", Status);
+        if (NT_SUCCESS(Status))
+            LdrUnloadDll(BaseAddress);
+    EndSeh(STATUS_SUCCESS);
+
+    StringCchPrintfW(szPath, _countof(szPath), L"%s\\advapi32", szSysDir);
+
+    for (pch = szPath; *pch != UNICODE_NULL; ++pch)
+    {
+        if (*pch == L'\\')
+            *pch = L'/';
+    }
+
+    RtlInitUnicodeString(&DllName, szPath);
+    StartSeh()
+        Status = LdrLoadDll(szWinDir, NULL, &DllName, &BaseAddress);
+        ok(Status == STATUS_SUCCESS, "Status = %lx\n", Status);
+        if (NT_SUCCESS(Status))
+            LdrUnloadDll(BaseAddress);
+    EndSeh(STATUS_SUCCESS);
+
+    StringCchPrintfW(szPath, _countof(szPath), L"%s\\advapi32.dll", szSysDir);
```

Comment:
```suggestion
    /* Test with only forward slashes in path; with file extension */
    StringCchPrintfW(szPath, _countof(szPath), L"%s\\advapi32.dll", szSysDir);
```

Reply:
But... there's a backslash right here in this line (same above)?

Reply:
Below code will convert.